### PR TITLE
Add len field to vectors

### PR DESCRIPTION
**Pull request by data-man**
Extracted from awesome #2945.

### CODE REVIEW

File: test/stage1/behavior/vector.zig [hunk 1]
```diff
@@ -250,3 +250,29 @@ test "initialize vector which is a struct field" {
...
 }
+
+test "vector member field access" {
+    const S = struct {
+        fn doTheTest() void {
+            const q: @Vector(4, u32) = undefined;
+            expect(q.len == 4);
+            const v = @Vector(4, i32);
+            expect(v.len == 4);
+            expect(v.bit_count == 32);
```
Comment (data-man) [inline test/stage1/behavior/vector.zig]: Should be v.bit_count = v[0].bit_count * v.len?

**Comment by data-man** [comment]
@andrewrk 
Please take a look.

**Comment by andrewrk** [comment]
This will not work because vectors of pointers to structs are supported, and `len` of the vector would conflict with a `len` field of a struct.

**Comment by data-man** [comment]
I didn't find any tests for vectors of pointers. :(

**Comment by data-man** [comment]
I don't know (yet) how to fix it, so feel free to close.

**Comment by andrewrk** [comment]
Yeah it's a tough problem. Sorry for the lack of organized docs that led to you spending time on this.

**Pull request closed without merging**